openpos-peripheral/src/main/java/org/jumpmind/pos/print/EscpPOSPrinter.java

```
@@ -132,6 +131,25 @@ public void openCashDrawer(String cashDrawerId) {
         printNormal(0, printerCommands.get(PrinterCommands.CASH_DRAWER_OPEN));
     }
 
+    @Override
+    public int waitForDrawerClose(long timeout) {
+        long startTime = System.currentTimeMillis();
+        int drawerState = 0;
+        try {
+            while (drawerState != 1 && System.currentTimeMillis() - startTime < timeout) {
+                if (System.currentTimeMillis() - startTime > timeout) {
+                    Thread.sleep(1000);
+                    getPeripheralConnection().getOut().write(new byte[]{0x1D, 0x72, 2});
+                    drawerState = getPeripheralConnection().getIn().read() == 1 ? 1 : 0;
+                }
+            }
+        } catch (Exception e) {
+            throw new PrintException("Failed to get drawer status");
```

This swallows the real exception

---

Removed the throw. Logging caught exception.